### PR TITLE
feat: [PL-39723]: EULA for Harness AIDA

### PR DESCRIPTION
**Pull request by mayankVermaHarness**
### Summary

JIRA: https://harness.atlassian.net/browse/PL-39723
EULA for Harness AIDA

### After Integrating EULA APIs

https://github.com/harness/harness-core-ui/assets/110456765/7d3c36e3-ae2c-46e4-b18c-9100b87d21ba

https://github.com/harness/harness-core-ui/assets/96057095/9185e149-98de-4a22-ab5a-13be17cece04

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

File: src/modules/23-default-settings/strings/strings.en.yaml [hunk 1]
```diff
@@ -39,3 +39,54 @@ gcpIncludeTaxes: (GCP Only) Include Taxes
...
+
+    1. Disclaimer. AIDA is offered "as is" without any warranties, expressed or implied and Harness makes no representations or warranties or guarantees regarding the accuracy, reliability, or performance of AIDA features or the Output. Output is based on patterns and probabilities, not certainty and may inadvertently be biased or unfair. Harness will not be responsible or liable for any errors, omissions, or inaccuracies in AIDA or the Output.
+
+    1. BETA SERVICES. UNLESS OTHERWISE COMMUNICATED BY HARNESS IN WRITING, AIDA IS A BETA SERVICE SUBJECT TO THE BETA SERVICES TERMS IN THE UNDERLYING TERMS.
+
+    1. Limitation of Liability. To the maximum extent permitted by applicable law, Harness shall not be liable for any direct, indirect, incidental, consequential, or special damages arising out of or in connection with the use or inability to use AIDA, including but not limited to the loss of profits, data, or business opportunities.
```
Comment (ishantmehtaharness) [inline src/modules/23-default-settings/strings/strings.en.yaml:70]: We have multiple `1. `
Is this expected? do we need proper numbering

Comment (mayankVermaHarness) [reply src/modules/23-default-settings/strings/strings.en.yaml:70]: Yes, this gets converted to proper numbering when we convert this markdown to HTML. Please check the video/screenshots in PR description.

File: package.json [hunk 2]
```diff
@@ -71,7 +71,7 @@
     "@harnessio/ff-react-client-sdk": "1.3.0",
     "@harnessio/react-audit-service-client": "^0.3.0",
     "@harnessio/react-idp-service-client": "0.12.0",
-    "@harnessio/react-ng-manager-client": "^1.8.0",
+    "@harnessio/react-ng-manager-client": "file:.yalc/@harnessio/react-ng-manager-client",
```
Comment (abhinavrastogi-harness) [inline package.json:74]: needs to be fixed

**Review by abhinavrastogi-harness — CHANGES_REQUESTED** [review]

File: src/modules/27-platform/strings/strings.en.yaml [hunk 3]
```diff
@@ -47,6 +47,59 @@ defaultSettings:
   exportServiceVariablesAsEnvVariables: Export Service Variables as Env Variables
   ticketingToolLabel: Ticketing Tool
   ticketingToolConnectorLabel: Ticketing Tool Connector
+  eula: End User License Agreement
+  aida:
+    aidaEULAText: |
+      **Harness AI Developer Assistant (AIDA) Terms**
```
Comment (abhinavrastogi-harness) [inline src/modules/27-platform/strings/strings.en.yaml:53]: Can we consider loading this text from Contentful?
It'll avoid committing the huge text to the strings framework, and allow docs team to change it without requiring a code deployment.

File: package.json (hunk 2 above)
Comment (namanharness) [reply package.json:74]: fixed

**Review by abhinavrastogi-harness — APPROVED** [review]

File: src/modules/27-platform/default-settings/utils/utils.ts [hunk 4]
```diff
@@ -9,6 +9,8 @@ import { StoreMetadata, StoreType } from '@common/constants/GitSyncTypes'
 import { SettingType } from '@common/constants/Utils'
 import { ResponseListSettingResponseDTO } from 'services/cd-ng'
 
+export const AidaAgreementType = 'AIDA'
```
Comment (mayankVermaHarness) [inline src/modules/27-platform/default-settings/utils/utils.ts:12]:
```suggestion
export const AgreementType = {
  AIDA: 'AIDA'
}
```

File: src/modules/27-platform/strings/strings.en.yaml (hunk 3 above)
Comment (mayankVermaHarness) [reply src/modules/27-platform/strings/strings.en.yaml:54]: As discussed, we will get this from Strapi.

File: src/modules/27-platform/default-settings/components/__test__/EulaComponents.test.tsx [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+        }}
+      >
+        <AIDASettingsRenderer {...props} />
+      </TestWrapper>
+    )
+    const input = await screen.findByTestId('aidaToggleStatus')
+    expect(input).toBeInTheDocument()
+
+    if (input) {
+      fireEvent.click(input)
```
Comment (mayankVermaHarness) [inline src/modules/27-platform/default-settings/components/__test__/EulaComponents.test.tsx:65]: For clicks, prefer using `userEvent.click`

File: src/modules/27-platform/default-settings/utils/utils.ts (hunk 4 above)
Comment (namanharness) [reply src/modules/27-platform/default-settings/utils/utils.ts:12]: image

Getting this when putting it in object

**Comment by mayankVermaHarness** [comment]
@namanharness I was the author of original PR, so I can't approve this.

**Review by namanharness — APPROVED** [review]

File: src/modules/27-platform/default-settings/components/__test__/EulaComponents.test.tsx (hunk 5 above)
Comment (namanharness) [reply src/modules/27-platform/default-settings/components/__test__/EulaComponents.test.tsx:65]: Will update in next PR

File: src/modules/27-platform/default-settings/utils/utils.ts (hunk 4 above)
Comment (namanharness) [reply src/modules/27-platform/default-settings/utils/utils.ts:12]: Will update in next PR

**Pull request merged**